src/main/webapp/app/exercises/programming/participate/programming-submission.service.ts

```diff
@@ -652,4 +656,40 @@ export class ProgrammingSubmissionService implements IProgrammingSubmissionServi
             }
         }
     }
+
+    /**
+     * Locks the given submission, exports it, transforms it to base64, and sends it to Orion
+     *
+     * @param exerciseId id of the exercise the submission belongs to
+     * @param submission submission to send to Orion
+     * @param correctionRound correction round
+     */
+    async downloadSubmissionInOrion(exerciseId: number, submission: Submission | 'new', correctionRound = 0) {
+        this.javaBridge.isCloning(true);
+        const submissionId: number =
+            submission === 'new'
+                ? (await this.getProgrammingSubmissionForExerciseForCorrectionRoundWithoutAssessment(exerciseId, true, correctionRound).toPromise()).id!
+                : submission.id!;
+        const exportOptions: RepositoryExportOptions = {
+            exportAllParticipants: false,
+            filterLateSubmissions: false,
+            addParticipantName: false,
+            combineStudentCommits: false,
+            anonymizeStudentCommits: true,
+            normalizeCodeStyle: false,
+            hideStudentNameInZippedFolder: true,
+        };
+        this.lockAndGetProgrammingSubmissionParticipation(submissionId, correctionRound).subscribe((programmingSubmission: ProgrammingSubmission) => {
+            this.repositoryExportService.exportReposByParticipations(exerciseId, [programmingSubmission.participation!.id!], exportOptions).subscribe((res: HttpResponse<Blob>) => {
+                const reader = new FileReader();
+                reader.readAsDataURL(res.body!);
+                reader.onloadend = () => {
+                    const result = reader.result as string;
+                    // remove prefix
+                    const base64data = result.substr(result.indexOf(',') + 1);
+                    this.javaBridge.downloadSubmission(submissionId, correctionRound, base64data);
+                };
```

Comment:
This might throw an error in case reading fails, that is usually catched. Similiar to onloadend, you can do sth like this:
https://developer.mozilla.org/en-US/docs/Web/API/FileReader/onerror